src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -625,5 +626,90 @@ public override async Task<VSInternalWrapWithTagResponse> RazorWrapWithTagAsync(
 
             return request?.Response;
         }
+
+        public override async Task<RazorFoldingRangeResponse> ProvideFoldingRangesAsync(RazorFoldingRangeRequestParam foldingRangeParams, CancellationToken cancellationToken)
+        {
+            if (foldingRangeParams is null)
+            {
+                throw new ArgumentNullException(nameof(foldingRangeParams));
+            }
+
+            var csharpRanges = new List<OmniSharp.Extensions.LanguageServer.Protocol.Models.FoldingRange>();
+            var csharpDocument = GetCSharpDocumentSnapshsot(foldingRangeParams.TextDocument.Uri.ToUri());
+            if (csharpDocument is not null)
+            {
+                var synchronized = await _documentSynchronizer.TrySynchronizeVirtualDocumentAsync(
+                    foldingRangeParams.DocumentHostVersion, csharpDocument, cancellationToken);
+
+                if (synchronized)
```

Comment:
What happens if we fail to synchronize and return `null` / empty folding ranges after having successfully returned folding ranges before? Does it clear all folding ranges or does it try again? Wondering if we have to worry about flashing at all